File: counting_sort/python/counting_sort.py
def counting_sort(arr, maximum_value):
    m = maximum_value + 1

    count = [0] * m

    for i in arr:
        count[i] += 1
    
    k = 0

    for i in range(m):
        for _ in range(count[i]):
            arr[k] = i
            k += 1

nums = [0, 2, 0, 1, 3, 4, 7, 1, 3, 2, 9, 3, 9, 10]

counting_sort(nums, 10)

print(nums)

File: counting_sort/javascript/counting_sort.js
const counting_sort = (arr, maximum_value) => {
    let buck = new Array(maximum_value + 1);
    let index = 0;
    let i;

    for(i = 0; i < arr.length; i++) {
        if(!buck[arr[i]]) {
            buck[arr[i]] = 0;
        }
        buck[arr[i]]++;
    }

    for(i = 0; i < buck.length; i++) {
        while(buck[i] > 0) {
            arr[index++] = i;
            buck[i]--;
        }
    }
}

let nums = [0, 2, 0, 1, 3, 4, 7, 1, 3, 2, 9, 3, 9, 10]

counting_sort(nums, 10)

console.log(nums)